src/SessionMessages/SessionMessages.tsx

```diff
@@ -45,6 +48,38 @@ interface SessionMessagesProps {
   children?: (conversations: Conversation[]) => ReactNode;
 }
 
+const executeScrollLogic = (
+  messagesRef: RefObject<HTMLDivElement>,
+  contentRef: RefObject<HTMLDivElement>,
+  mutationObserver: MutationObserver
+) => {
+  if (contentRef.current) {
+    const atBottom =
+      contentRef.current.scrollHeight - contentRef.current.clientHeight ===
+      contentRef.current.scrollTop;
+    if (atBottom) {
+      // If we are at the bottom, don't scroll
+      mutationObserver.disconnect();
+      return;
+    }
+  }
+  if (messagesRef.current) {
+    const lastMessage = messagesRef.current
+      .lastElementChild as HTMLElement | null;
+
+    if (lastMessage) {
+      lastMessage.scrollIntoView({
+        behavior: 'smooth',
+        block: 'start'
+      });
+    }
+    // Disconnect the mutation observer after the scroll
+    mutationObserver.disconnect();
+  }
+};
+
+const debouncedExecuteScrollLogic = debounce(executeScrollLogic, 100);
```

Comment:
If there was multiple chatbots on the page, this would be an issue. We need to move this to be instanced.